File: examples/typescript/basic.ts
import { EchoZeroClient, signInboundWebhook } from '../../packages/typescript/src/index.js';

const client = new EchoZeroClient({
  baseUrl: process.env.ECHOZERO_BASE_URL ?? 'https://mcp.echozero.app',
  apiKey: process.env.ECHOZERO_API_KEY,
  bearerToken: process.env.ECHOZERO_OAUTH_TOKEN,
});

const me = await client.get('/api/v1/users/me');
console.log(me);

if (process.env.ECHOZERO_WEBHOOK_SECRET) {
  const body = { text: 'BUY SOL 500 USDC' };
  const headers = await signInboundWebhook({
    signingSecret: process.env.ECHOZERO_WEBHOOK_SECRET,
    body,
  });
  console.log(headers);
}

File: examples/python/basic.py
import os

from echozero import EchoZeroClient, sign_inbound_webhook


client = EchoZeroClient(
    base_url=os.getenv("ECHOZERO_BASE_URL", "https://mcp.echozero.app"),
    api_key=os.getenv("ECHOZERO_API_KEY"),
    bearer_token=os.getenv("ECHOZERO_OAUTH_TOKEN"),
)

print(client.get("/api/v1/users/me"))

webhook_secret = os.getenv("ECHOZERO_WEBHOOK_SECRET")
if webhook_secret:
    body = {"text": "BUY SOL 500 USDC"}
    print(sign_inbound_webhook(signing_secret=webhook_secret, body=body))
